R/generateFeatureImportance.R

```diff
@@ -0,0 +1,144 @@
+#'
+#' @template arg_task
+#' @param method [\code{character(1)}]\cr
+#'   The method used to compute the feature importance.
+#'   Default is \dQuote{permutation.importance}.
+#' @template arg_learner
+#' @param features [\code{character}]\cr
+#'   The features to compute the importance of.
+#'   The default is all of the features contained in the \code{\link{Task}}.
+#' @param interaction [\code{logical(1)}]\cr
+#'   Whether to compute the importance of the \code{features} argument jointly.
+#'   The default is \code{FALSE}.
+#' @template arg_measure
+#' @param contrast [\code{function}]\cr
+#'   A difference function that takes a numeric vector and returns a numeric vector
+#'   of the same length.
+#'   The default is element-wise difference between the vectors.
+#' @param aggregation [\code{function}]\cr
+#'   A function which aggregates the differences.
+#'   This function must take a numeric vector and return a numeric vector of length 1.
+#'   The default is \code{mean}.
+#' @param nperm [\code{integer(1)}]\cr
+#'   The number of Monte-Carlo iterations to use in computing the feature importance.
+#'   The default is 50.
+#' @param replace [\code{logical(1)}]\cr
+#'   Whether or not to sample the feature values with or without replacement.
+#'   The default is \code{TRUE}.
+#' @param local [\code{logical(1)}]\cr
+#'   Whether to compute the per-observation importance.
+#'   The default is \code{FALSE}.
+#'
+#' @return [\code{FeatureImportance}]. A named list which contains the computed feature importance and the input arguments.
+#' @export
+generateFeatureImportanceData = function(task, method = "permutation.importance",
+  learner, features = getTaskFeatureNames(task), interaction = FALSE, measure,
+  contrast = function(x, y) x - y, aggregation = mean, nperm = 50L, replace = TRUE,
+  local = FALSE) {
+  
+  learner = checkLearner(learner)
+  measure = checkMeasures(measure, learner)
+  if (length(measure) > 1L)
+    stop("only one measure is allowed.")
+  if (getTaskType(task) != learner$type)
+    stopf("Expected task of type '%s', not '%s'", getTaskType(task), learner$type)
+  assertCount(nperm)
+  test.contrast = contrast(1, 1)
+  if (!(is.numeric(test.contrast)))
+    stop("the contrast function must return a numeric vector.")
+  if (!length(test.contrast) == 1L)
+    stop("the contrast function must return a numeric vector the same length as the input.")
+  test.aggregation = aggregation(1:2)
+  if (!is.numeric(test.aggregation))
+    stop("aggregation argument doesn't return a numeric vector.")
+  if (!(length(test.aggregation) == 1L))
+    stop("aggregation function must either return 1 number or a numeric vector of the same length as the number of rows in the task data.frame.")
+
+  out = switch(method,
+    "permutation.importance" = doPermutationImportance(
+      task, learner, features, interaction, measure, contrast, aggregation, nperm, replace, local)
+    )
+
+  makeS3Obj(
+    "FeatureImportance",
+    data = out,
```

Comment:
I would prefer "res" instead of "data" here to make it clearer that this is the result of the operation.